### PR TITLE
[kube-state-metrics] add indentation for customLabels in role template

### PR DESCRIPTION
**Pull request by ecthelion77**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-state-metrics/Chart.yaml [hunk 1]
```diff
@@ -2,18 +2,18 @@ apiVersion: v2
 name: kube-state-metrics
 description: Install kube-state-metrics to generate and expose cluster-level metrics
 keywords:
-- metric
-- monitoring
-- prometheus
-- kubernetes
+  - metric
```
Comment (mrueg) [inline charts/kube-state-metrics/Chart.yaml:5]: Can you remove the formatting changes here?

File: charts/kube-state-metrics/templates/role.yaml [hunk 2]
```diff
@@ -11,14 +11,14 @@ metadata:
...
     app.kubernetes.io/managed-by: {{ $.Release.Service }}
     app.kubernetes.io/version: "{{ $.Chart.Version }}"
-    app.kubernetes.io/part-of: {{ template "kube-state-metrics.name" $ }} 
-    helm.sh/chart: {{ $.Chart.Name }}-{{ $.Chart.Version }} 
+    app.kubernetes.io/part-of: {{ template "kube-state-metrics.name" $ }}
+    helm.sh/chart: {{ $.Chart.Name }}-{{ $.Chart.Version }}
     {{- if $.Values.customLabels }}
-    {{ toYaml $.Values.customLabels }}
-    {{- end }} 
+    {{ toYaml $.Values.customLabels | nindent 4 }}
```
Comment (mrueg) [inline charts/kube-state-metrics/templates/role.yaml:20]: Just to make sure, did you have the chance to test it? :)

File: charts/kube-state-metrics/Chart.yaml [hunk 3]
```diff
@@ -7,13 +7,13 @@ keywords:
...
 home: https://github.com/kubernetes/kube-state-metrics/
 sources:
-- https://github.com/kubernetes/kube-state-metrics/
+  - https://github.com/kubernetes/kube-state-metrics/
 maintainers:
-- name: tariq1890
-  email: tariq.ibrahim@mulesoft.com
-- name: mrueg
-  email: manuel@rueg.eu
+  - name: tariq1890
```
Comment (mrueg) [inline charts/kube-state-metrics/Chart.yaml:16]: Remove these here as well, only raise the chart version.

File: charts/kube-state-metrics/Chart.yaml [hunk 4]
```diff
@@ -7,13 +7,13 @@ keywords:
 - prometheus
 - kubernetes
 type: application
-version: 4.1.0
+version: 4.1.1
 appVersion: 2.2.4
 home: https://github.com/kubernetes/kube-state-metrics/
 sources:
-- https://github.com/kubernetes/kube-state-metrics/
+  - https://github.com/kubernetes/kube-state-metrics/
```
Comment (mrueg) [inline charts/kube-state-metrics/Chart.yaml:14]: Remove these here as well, only raise the chart version.

File: charts/kube-state-metrics/templates/role.yaml (hunk 2 above)
Comment (ecthelion77) [reply charts/kube-state-metrics/templates/role.yaml:20]: i'm doing that and push again when it's done

Comment (ecthelion77) [reply charts/kube-state-metrics/templates/role.yaml:20]: without customLabels :
````
Name:         stsdiscovery-dev01-208-monitoring-kube-state-metrics
Labels:       app.kubernetes.io/component=metrics
              app.kubernetes.io/instance=dev01-208-monitoring
              app.kubernetes.io/managed-by=Helm
              app.kubernetes.io/name=kube-state-metrics
              app.kubernetes.io/part-of=kube-state-metrics
              app.kubernetes.io/version=2.2.4
              argocd.argoproj.io/instance=dev01-208-monitoring
              helm.sh/chart=kube-state-metrics-4.1.0
Annotations:  <none>
PolicyRule:
  Resources          Non-Resource URLs  Resource Names                             Verbs
  ---------          -----------------  --------------                             -----
  statefulsets.apps  []                 [dev01-208-monitoring-kube-state-metrics]  [get list watch]
  pods               []                 []                                         [get]
````

with a custom Label :
````
Name:         stsdiscovery-dev01-208-monitoring-kube-state-metrics
Labels:       app.kubernetes.io/component=metrics
              app.kubernetes.io/instance=dev01-208-monitoring
              app.kubernetes.io/managed-by=Helm
              app.kubernetes.io/name=kube-state-metrics
              app.kubernetes.io/part-of=kube-state-metrics
              app.kubernetes.io/version=2.2.4
              argocd.argoproj.io/instance=dev01-208-monitoring
              helm.sh/chart=kube-state-metrics-4.1.1
              testCustomLabel=my-custom-label-for-test
Annotations:  <none>
PolicyRule:
  Resources          Non-Resource URLs  Resource Names                             Verbs
  ---------          -----------------  --------------                             -----
  statefulsets.apps  []                 [dev01-208-monitoring-kube-state-metrics]  [get list watch]
  pods               []                 []                                         [get]
````

Comment (ecthelion77) [reply charts/kube-state-metrics/templates/role.yaml:20]: @mrueg so normally files are correctly formatted, tests are done and working, can you validate the PR

**Comment by mrueg** [comment]
Thanks for your contribution!

**Pull request merged**